app/pdf_redactor.py

```diff
@@ -551,16 +550,20 @@ def toUnicode(string, font, fontcache):
             fontcache[font.ToUnicode.stream] = CMap(font.ToUnicode)
         cmap = fontcache[font.ToUnicode.stream]
 
-        string = cmap.decode(string)
-        # print(string, end='', file=sys.stderr)
-        # sys.stderr.write(string)
-        return string
-    elif get_encoding(font) == "/WinAnsiEncoding":
+        try:
+            return cmap.decode(string)
+        except RedactionException:
+            # if there's a redaction exception, try and decode using fall back methods.
```

Comment:
do we still want to add a logging event here? Or do you think there is no need since we are rescuing it pretty cleanly? :)

Reply:
Yeah, that was my thinking. In practice this line executes for every text token, and in those rother PDFs we were often seeing four or five log lines.

We think it'll rescue cleanly, but if it doesn't:

* if the encoding lines fail (because it's got WinAnsiEncoding but it expects ToUnicode to work for some reason) and it for example instead encodes badly and returns gibberish, then we won't match, and will log `"No matches for address block during redaction procedure"`
* if it doesn't have Win or Mac encoding, then it'll log `f'Unrecognised font with encoding {font.Encoding}, may not be able to redact properly'`, return a `?`, and we'll log that No Matches log message again

so both are things we can pick up